src/components/AvatarWithIndicator.js

```diff
@@ -14,37 +19,116 @@ const propTypes = {
+            this.startSyncIndicator();
+        }
+    }
+
+    componentDidUpdate(prevProps) {
+        if (!prevProps.isSyncing && this.props.isSyncing) {
+            this.startSyncIndicator();
+        } else if (prevProps.isSyncing && !this.props.isSyncing) {
+            this.stopSyncIndicator();
+        }
+    }
+
+    componentWillUnmount() {
+        this.stopSyncIndicator();
+    }
+
+    startSyncIndicator() {
+        Animated.loop(Animated.timing(this.rotate, {
+            toValue: 1,
+            duration: 3000,
+            easing: Easing.linear,
+            isInteraction: false,
+            useNativeDriver: true,
+        })).start();
+        Animated.spring(this.scale, {
+            toValue: 1.35,
+            tension: 1,
+            isInteraction: false,
+            useNativeDriver: true,
+        }).start();
+    }
+
+    stopSyncIndicator() {
+        Animated.spring(this.scale, {
+            toValue: 1,
+            tension: 1,
+            isInteraction: false,
+            useNativeDriver: true,
+        }).start(() => {
+            this.rotate.resetAnimation();
+            this.scale.resetAnimation();
+            this.rotate.setValue(0);
+        });
+    }
+
+    render() {
+        const indicatorStyles = [
+            styles.alignItemsCenter,
+            styles.justifyContentCenter,
+            {
+                transform: [{
+                    rotate: this.rotate.interpolate({
+                        inputRange: [0, 1],
+                        outputRange: ['0deg', '-360deg'],
+                    }),
+                }, {
+                    scale: this.scale,
+                }],
+            },
+            this.props.isSyncing ? styles.statusIndicatorSyncing : null,
+            this.props.size === 'large' ? styles.statusIndicatorLarge : styles.statusIndicator,
+            this.props.isActive ? styles.statusIndicatorOnline : styles.statusIndicatorOffline,
+        ];
+
+        return (
+            <View
+                style={[this.props.size === 'large' ? styles.avatarLarge : styles.sidebarAvatar]}
+            >
+                <Avatar
+                    style={[this.props.size === 'large' ? styles.avatarLarge : null]}
+                    source={this.props.source}
+                />
+                <Animated.View style={StyleSheet.flatten(indicatorStyles)}>
+                    {this.props.isSyncing && (
+                        <Icon
+                            src={Sync}
+                            fill={themeColors.textReversed}
+                            width="100%"
+                            height="100%"
+                        />
+                    )}
+                </Animated.View>
+            </View>
+        );
+    }
+}
 
 AvatarWithIndicator.defaultProps = defaultProps;
 AvatarWithIndicator.propTypes = propTypes;
 AvatarWithIndicator.displayName = 'AvatarWithIndicator';
-export default memo(AvatarWithIndicator);
+export default AvatarWithIndicator;
```

Comment:
NAB: For my understanding, you got rid of the memo here because you're animating this right? I'm guessing when you were testing this it wouldn't wouldn't animate while this was wrapping.

Reply:
No I used PureComponent and we don't need it now.